fastpay/src/network.rs

```diff
@@ -140,53 +140,60 @@ impl MessageHandler for RunningServerState {
                             .map(|info| Some(serialize_info_response(&info))),
-                        SerializedMessage::Cert(message) => {
+                        SerializedMessage::Confirmation(message) => {
                             let confirmation_order = ConfirmationOrder {
-                                transfer_certificate: message.as_ref().clone(),
+                                transfer_certificate: *message,
                             };
                             match self
                                 .server
                                 .state
                                 .handle_confirmation_order(confirmation_order)
                             {
-                                Ok((info, send_shard)) => {
-                                    // Send a message to other shard
-                                    if let Some(cross_shard_update) = send_shard {
-                                        let shard = cross_shard_update.shard_id;
-                                        let tmp_out = serialize_cross_shard(&message);
-                                        debug!(
-                                            "Scheduling cross shard query: {} -> {}",
-                                            self.server.state.shard_id, shard
-                                        );
-                                        self.cross_shard_sender
-                                            .send((tmp_out, shard))
-                                            .await
-                                            .expect("internal channel should not fail");
-                                    };
+                                Ok((info, continuation)) => {
+                                    self.handle_continuation(continuation).await;
 
                                     // Response
                                     Ok(Some(serialize_info_response(&info)))
                                 }
                                 Err(error) => Err(error),
                             }
                         }
-                        SerializedMessage::InfoReq(message) => self
+                        SerializedMessage::InfoRequest(message) => self
                             .server
                             .state
                             .handle_account_info_request(*message)
                             .map(|info| Some(serialize_info_response(&info))),
-                        SerializedMessage::CrossShard(message) => {
-                            match self
-                                .server
-                                .state
-                                .handle_cross_shard_recipient_commit(*message)
-                            {
-                                Ok(_) => Ok(None), // Nothing to reply
+                        SerializedMessage::CrossShardRequest(request) => {
+                            use CrossShardRequest::*;
```

Comment:
Import at the top of the file?

Reply:
Actually, you could make the case that `use Foo:::*` is not great at the top of the file. (I know we do it everywhere now but we shouldn't.)